pkg/controllers/manila/openstack.go

```diff
@@ -80,19 +80,19 @@ func (o *openStackClient) GetShareTypes() ([]sharetypes.ShareType, error) {
 
 	err = openstack.Authenticate(provider, *opts)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("cannot authenticate with given credentials: %w", err)
 	}
 
 	client, err := openstack.NewSharedFileSystemV2(provider, gophercloud.EndpointOpts{
```

Comment:
`GetShareTypes()` does many things. Is it possible that say `404` from `NewSharedFileSystemV2` needs to be handled differently than `404` from e.g. `sharetypes.List` below?

The operator will get disabled with the same message in both cases.

Reply:
It's a good question. My thoughts are:
- ErrEndpointNotFound can come from NewSharedFileSystemV2 only;
- ErrDefault403 always comes from sharetypes.List, Keystone doesn't return this code for the authentication request;
- ErrDefault404 potentially can be returned both from Keystone and Manila, but frankly speaking it is an absolutely extreme scenario for Keystone and I can't imagine a situation that may lead to this.